src/arrow2/src/io/parquet/read/deserialize/dictionary/nested.rs

```diff
@@ -184,7 +184,11 @@ pub fn next_dict<K: DictionaryKey, I: Pages, F: Fn(&DictPage) -> Box<dyn Array>>
                 init,
                 items,
                 None,
-                remaining,
+                (remaining, &mut 0), // TODO(issue#2537): Daft does not currently support Arrow's
```

Comment:
Would it make more sense to panic here instead?

Reply:
I'm sweating a little remembering my kernel days with Greg KH telling off the email thread for using WARN instead of throwing an error.

But since we expect this code to be unreachable for now I guess it's fine. Changed this to panic

Reply:
LOL! Yeah I think it's better for now to be explict and have the user see the error rather than return weird results